### PR TITLE
Fixed #24944 -- Added extra_email_context parameter to password_reset() view.

### PR DESCRIPTION
**Pull request by SujaySKumar**

### CODE REVIEW

File: django/contrib/auth/views.py [hunk 1]
```diff
@@ -179,6 +179,10 @@ def password_reset(request, is_admin_site=False,
                     RemovedInDjango20Warning, 3
                 )
                 opts = dict(opts, domain_override=request.get_host())
+
```
Comment (MoritzS) [inline django/contrib/auth/views.py]: This line contains spaces that flake8 complains about.

File: docs/topics/auth/default.txt [hunk 2]
```diff
@@ -1246,7 +1246,11 @@ implementation details see :ref:`using-the-views`.
     * ``token``: Token to check that the reset link is valid.
 
     Sample ``registration/password_reset_email.html`` (email body template):
-
+
```
Comment (MoritzS) [inline docs/topics/auth/default.txt]: Remove all trailing whitespaces.

**Comment by MoritzS** [comment]
The trailing whitespaces in the docs are still there.

**Comment by timgraham** [comment]
What I had in mind was adding an `extra_email_context` parameter to the `password_reset` view.

**Comment by timgraham** [comment]
Like this:
```
def password_reset(request, is_admin_site=False,
                   ...
                   html_email_template_name=None,
                   extra_email_context=None):
```
Also needs tests. See also our [patch review checklist](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/submitting-patches/#patch-review-checklist).

**Comment by SujaySKumar** [comment]
Hi Tim,

Is this ok?
I'm sorry if you find me a little frustrating. I am new to open source
contribution.
I am working on the tests for this feature. I will commit it as soon as it
is done.
Thanks!

Regards,
Sujay S Kumar

File: django/contrib/auth/views.py [hunk 3]
```diff
@@ -179,6 +180,10 @@ def password_reset(request, is_admin_site=False,
                     RemovedInDjango110Warning, 3
                 )
                 opts = dict(opts, domain_override=request.get_host())
+
+            if extra_context is not None:
```
Comment (timgraham) [inline django/contrib/auth/views.py]:
```python
if extra_email_context is not None:
    opts.update(extra_email_context)
````

**Comment by timgraham** [comment]
Please send a new PR if you continue work on this, thanks!

**Comment by SujaySKumar** [comment]
Hi Tim,
I have sent a new pull request. Please check it out. Sorry for the delay.

**Pull request closed without merging**